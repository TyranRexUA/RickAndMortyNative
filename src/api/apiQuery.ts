import gql from "graphql-tag"

export const getEpisodesQuery = gql`
    query getEpisodesQuery($page: Int!, $name: String){
        episodes(page: $page, filter: {name: $name}) {
            info {
                count
                pages
                next
                prev
            }
            results {
                name
                id
            }
        }
    }
`

export const getSingleEpisodeQuery = gql`
    query getEpisodesQuery($id: ID!){
        episode(id: $id) {
            id
            name
            air_date
            characters {
                id
                name
                image
            }
        }
    }
`

export const getCharactersQuery = gql`
    query getEpisodesQuery($page: Int!){
        characters(page: $page) {
            info {
                count
                pages
                next
                prev
            }
            results {
                name
                id
            }
        }
    }
`

export const getSingleCharacterQuery = gql`
    query getEpisodesQuery($id: ID!){
        character(id: $id) {
            id
            name
            image
            species
            gender
            episode {
                name
                id
            }
        }
    }
`