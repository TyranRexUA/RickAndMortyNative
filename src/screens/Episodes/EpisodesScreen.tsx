
import React from 'react';
import { View } from 'react-native';
import EpisodesList from './EpisodesList';
import styles from './styles/EpisodesScreenStyle';

const EpisodesScreen = () => {
    return (
        <View style={styles.screenContainer}>
            <EpisodesList />
        </View>
    )
}

export default EpisodesScreen;