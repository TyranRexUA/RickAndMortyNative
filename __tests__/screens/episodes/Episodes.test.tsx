import React from 'react';
import { render, act, fireEvent, cleanup, RenderAPI } from '@testing-library/react-native';
import EpisodesScreen from '../../../src/screens/Episodes/EpisodesScreen';
import { MockedProvider } from '@apollo/client/testing';
import { getEpisodesQuery } from '../../../src/api/apiQuery';
import { getEpisodesPage1, getEpisodesPage2, getEpisodesWithFilter } from '../../../__mocks__/getEpisodes';
import {mockedNavigate} from '../../settings/setupTests';

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: mockedNavigate,
        dispatch: jest.fn(),
      }),
    };
  });

describe('EpisodesItem', () => {
    let wrapper: RenderAPI;
    const promise = Promise.resolve()

    const episodesMockPage1 = {
        request: {
            query: getEpisodesQuery,
            variables: { "page": 1, "name": "" },
        },
        result: getEpisodesPage1,
    };

    const episodesMockPage2 = {
        request: {
            query: getEpisodesQuery,
            variables: { "page": 2, "name": "" },
        },
        result: getEpisodesPage2,
    }

    const episodesMockWithFilterName = {
        request: {
            query: getEpisodesQuery,
            variables: { "page": 1, "name": "ab" },
        },
        result: getEpisodesWithFilter,
    }

    const errorMock = {
        request: {
            query: getEpisodesQuery,
            variables: { "page": 1, "name": "error" },
        },
        error: new Error(),
    }

    afterEach(cleanup)

    beforeEach(async () => {
        wrapper = render(
            <MockedProvider mocks={[episodesMockPage1, episodesMockPage2, episodesMockWithFilterName, errorMock]}>
                <EpisodesScreen />
            </MockedProvider>
        );
        await act(async () => promise)
    });

    test('with loader', async () => {
        await act(async () => {
            expect(wrapper.getByTestId('Loader')).toBeTruthy();
        })
    })

    test('With error', async () => {
        await act(async () => {
            await fireEvent(wrapper.getByTestId('TextInput'), 'onChangeText', 'error')
        })
        expect(wrapper.getByText(/not found/i)).toBeTruthy();
    })

    describe('with data', () => {
        const scroll = {
            nativeEvent: {
                contentSize: {
                    height: 200
                },
                layoutMeasurement: {
                    height: 50
                },

                contentOffset: {
                    y: 175,
                },
            },
        }

        beforeEach(async () => {
            await act(() => promise)
        });

        test('check Episodes', async () => {
            expect(wrapper.getAllByTestId('EpisodesItem').length).toBe(20);
        });

        test('check navigation', async () => {
            await act(async () => {
                await fireEvent(wrapper.getAllByTestId('EpisodesItem')[0], 'onPress')
            })
            expect(mockedNavigate).toHaveBeenCalled();
        })

        test('check loadMore Episodes', async () => {
            await act(async () => {
                await fireEvent.scroll(wrapper.getByTestId('ScrollView'), scroll)
            })
            expect(wrapper.getAllByTestId('EpisodesItem').length).toBe(40);
            await act(async () => {
                await fireEvent.scroll(wrapper.getByTestId('ScrollView'), scroll)
            })
            expect(wrapper.getAllByTestId('EpisodesItem').length).toBe(40);
        })

        describe('check name Filter', () => {
            let inputText: React.ReactElement<{value: string}>;

            beforeEach(async () => {
                inputText = wrapper.getByTestId('TextInput');
                await act(async () => {
                    await fireEvent(inputText, 'onChangeText', 'ab')
                })
            });

            test('check value in TextInput', async () => {
                expect(inputText.props.value).toBe('ab');
            })

            test('check Episodes', async () => {
                expect(wrapper.getAllByTestId('EpisodesItem').length).toBe(2);
            })

            test('check clearInputText', async () => {
                await act(async () => {
                    await fireEvent(wrapper.getByTestId('clearInputText'), 'press')
                })
                expect(wrapper.getAllByTestId('EpisodesItem').length).toBe(20);
                expect(inputText.props.value).toBe('');
            })
        })

        test('Snapshot checking', async () => {
            expect(wrapper.toJSON()).toMatchSnapshot();
        })
    })

});