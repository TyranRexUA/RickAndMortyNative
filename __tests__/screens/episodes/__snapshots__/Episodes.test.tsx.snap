// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodesItem with data Snapshot checking 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 2,
          "borderTopRightRadius": 2,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          Object {
            "minHeight": 64,
          },
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <Text
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "rgb(250, 0, 0)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Search episodes name...
        </Text>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Search episodes name...
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        selectionColor="rgb(250, 0, 0)"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "zIndex": 1,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 64,
            },
            Object {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            Array [
              Object {},
            ],
          ]
        }
        testID="TextInput"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <RCTScrollView
    onScroll={[Function]}
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 10,
      }
    }
    testID="ScrollView"
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            1
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Pilot
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            2
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Lawnmower Dog
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            3
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Anatomy Park
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            4
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            M. Night Shaym-Aliens!
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            5
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Meeseeks and Destroy
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            6
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Rick Potion #9
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            7
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Raising Gazorpazorp
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            8
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Rixty Minutes
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            9
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Something Ricked This Way Comes
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            10
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Close Rick-counters of the Rick Kind
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            11
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Ricksy Business
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            12
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            A Rickle in Time
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            13
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Mortynight Run
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            14
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Auto Erotic Assimilation
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            15
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Total Rickall
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            16
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Get Schwifty
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            17
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            The Ricks Must Be Crazy
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            18
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Big Trouble in Little Sanchez
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            19
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Interdimensional Cable 2: Tempting Fate
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "whitesmoke",
            "display": "flex",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="EpisodesItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "padding": 5,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginRight": 10,
              }
            }
          >
            20
          </Text>
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Look Who's Purging Now
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
